src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -474,6 +478,8 @@ private function getValueType(Type $baseType, int $flags, bool $matchesAll): Typ
 		$offsetType = IntegerRangeType::fromInterval(0, null);
 		if ($this->containsUnmatchedAsNull($flags, $matchesAll)) {
 			$valueType = TypeCombinator::addNull($valueType);
+		}
+		if ($this->containsUnmatchedAsNull($flags, $matchesAll) || $matchesAll) {
 			// unmatched groups return -1 as offset
 			$offsetType = IntegerRangeType::fromInterval(-1, null);
```

Comment:
`preg_match_all` can have -1 in results, even without `PREG_UNMATCHED_AS_NULL`

https://3v4l.org/Lo7mk